src/Microsoft.ML.FastTree/Representation/TreeRegressorCollection.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Trainers.FastTree.Internal;
+
+namespace Microsoft.ML.FastTree.Representation
+{
+    /// <summary>
+    /// A list of <see cref="TreeRegressor"/>. To compute the output value of a <see cref="TreeRegressorCollection"/>, we need to compute
+    /// the output values of all trees in <see cref="Trees"/>, scale those values via <see cref="TreeWeights"/>, and finally sum the scaled
+    /// values and <see cref="Bias"/> up.
+    /// </summary>
+    public class TreeRegressorCollection
+    {
+        /// <summary>
+        /// It's a best friend for being accessed from LightGBM.
+        /// </summary>
+        [BestFriend]
+        internal readonly TreeEnsemble TreeEnsemble;
+
+        /// <summary>
+        /// When doing prediction, this is a value added to the weighted sum of all <see cref="Trees"/>' outputs.
+        /// </summary>
+        public double Bias { get; }
+
+        /// <summary>
+        /// <see cref="TreeWeights"/>[i] is the i-th <see cref="TreeRegressor"/>'s weight in this <see cref="TreeRegressorCollection"/>.
+        /// </summary>
+        public IReadOnlyList<double> TreeWeights { get; }
```

Comment:
>IReadOnlyList<double> [](start = 15, length = 21)

Here we are using `IReadOnlyList`. Sometimes we are using `ReadOnlySpan`. How about we pick *one*? (I have something of a preference for `IReadOnlyList` since spans are stack-only concepts and it's unclear to me that that is appropriate in this case.)

Reply:
No problem. IReadOnlyList will be the universal way to access enumerable attributes in Tree and TreeEnsemble.

---
In reply to: [252499854](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252499854) [](ancestors = 252499854)